### PR TITLE
Rewardの実装

### PR DESCRIPTION
**Pull request by myxyy**
## 概要

https://github.com/MLShukai/PrimitiveAMI/issues/36

## 変更内容

* 抽象クラスReward追加
* 継承クラスCuriosityReward追加
* テスト追加

## Submit前の確認項目

- [x] タイトルは一目でわかるようにし、説明文は PR を簡潔に説明するようにしましたか?
- [x] あなたの PR が、異なる変更を束ねたものではなく、ひとつのことだけを行うものであることを確認しましたか?
- [x] この PR で導入されたすべての変更点をリストアップしましたか?
- [x] PR を`make test`コマンドでローカルにテストしましたか?
- [x] `make format`コマンドで `pre-commit` フックを実行しましたか?

### CODE REVIEW

File: src/models/components/reward/curiosity_reward.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from .reward import Reward
+
+
+class CuriosityReward(Reward):
+    def __init__(self):
+        super().__init__()
+        self.mseloss = nn.MSELoss(reduction="none")
+
+    def reward(self, obs_hat: Tensor, obs: Tensor) -> Tensor:
+        return self.mseloss(obs_hat, obs).mean(dim=-1)
```
Comment (Geson-anko) [inline src/models/components/reward/curiosity_reward.py:13]: スカラーで返すのでdimの指定はいらないと思います。
```suggestion
        return self.mseloss(obs_hat, obs).mean()
```

File: tests/models/components/reward/test_curiosity_reward.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import pytest
+import torch
+from torch.distributions.distribution import Distribution
```
Comment (Geson-anko) [inline tests/models/components/reward/test_curiosity_reward.py:3]: コピペは、危険...
```suggestion
```

File: tests/models/components/reward/test_curiosity_reward.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    batch,
+    dim,
+    """,
+    [
+        (4, 623),
+        (32, 64),
+        (17, 234),
+    ],
+)
+def test_tanh_normal_stochastic_policy(batch, dim):
```
Comment (Geson-anko) [inline tests/models/components/reward/test_curiosity_reward.py:19]: 名前を修正
```suggestion
def test_curiosity_reward(batch, dim):
```

**Review by Geson-anko — CHANGES_REQUESTED** [review]
@myxyy 正直 `Reward`モジュールはパラメータモジュールじゃないので`nn.Module`を継承しなくていいかなって思ってるんですけど、思想的には、悪くないかなと思ってます。例えば ObsEncoderを内包している場合とかも想定できるのでいいかなーと。ちょっとした名前ミスあるので直しておいてください。

File: tests/models/components/reward/test_curiosity_reward.py (hunk 3 above)
Comment (myxyy) [reply tests/models/components/reward/test_curiosity_reward.py]: 修正しました https://github.com/MLShukai/PrimitiveAMI/pull/40/commits/ebed8b020ece37b621f71abc8cef0d1f5e62e800

File: tests/models/components/reward/test_curiosity_reward.py (hunk 2 above)
Comment (myxyy) [reply tests/models/components/reward/test_curiosity_reward.py]: 修正しました https://github.com/MLShukai/PrimitiveAMI/pull/40/commits/ebed8b020ece37b621f71abc8cef0d1f5e62e800

**Review by Geson-anko — APPROVED** [review]
@myxyy OK!
